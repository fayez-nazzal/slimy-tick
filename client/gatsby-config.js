module.exports = {
  siteMetadata: {
    title: `Slimy Tick`,
    description: `Manage your time in a modern way.`,
    author: `@fayez-nazza;`,
  },
  plugins: [
    `gatsby-plugin-react-helmet`,
    `gatsby-plugin-image`,
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `images`,
        path: `${__dirname}/src/images`,
      },
    },
    `gatsby-transformer-sharp`,
    `gatsby-plugin-sharp`,
    {
      resolve: `gatsby-plugin-manifest`,
      options: {
        name: `gatsby-starter-default`,
        short_name: `starter`,
        start_url: `/`,
        background_color: `#663399`,
        theme_color: `#663399`,
        display: `minimal-ui`,
        icon: `src/images/background.jpg`,
      },
    },
    {
      resolve: "gatsby-plugin-react-svg",
      options: {
        rule: {
          include: /svg/,
        },
      },
    },
    `gatsby-plugin-gatsby-cloud`,
    `gatsby-plugin-offline`,
  ],
}
