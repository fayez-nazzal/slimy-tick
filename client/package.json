{
  "name": "gatsby-starter-default",
  "private": true,
  "description": "A simple starter to get up and developing quickly with Gatsby",
  "version": "0.1.0",
  "author": "Kyle Mathews <mathews.kyle@gmail.com>",
  "dependencies": {
    "@apollo/client": "^3.3.15",
    "@date-io/moment": "1.x",
    "@material-ui/core": "^4.11.3",
    "@material-ui/icons": "^4.11.2",
    "@material-ui/pickers": "^3.3.10",
    "@reach/router": "^1.3.4",
    "@reduxjs/toolkit": "^1.5.1",
    "@testing-library/react": "^11.2.6",
    "apollo-link-context": "^1.0.20",
    "clsx": "^1.1.1",
    "draft-js": "^0.11.7",
    "eslint": "^5.16.0",
    "eslint-config-airbnb": "^18.2.1",
    "eslint-plugin-react": "^7.21.5",
    "gatsby": "^3.1.2",
    "gatsby-background-image": "^1.5.1",
    "gatsby-plugin-gatsby-cloud": "^2.1.0",
    "gatsby-plugin-image": "^1.1.2",
    "gatsby-plugin-manifest": "^3.1.0",
    "gatsby-plugin-offline": "^4.1.0",
    "gatsby-plugin-react-helmet": "^4.1.0",
    "gatsby-plugin-react-svg": "^3.0.1",
    "gatsby-plugin-sharp": "^3.1.2",
    "gatsby-source-filesystem": "^3.1.0",
    "gatsby-transformer-sharp": "^3.1.0",
    "graphql": "^15.5.0",
    "isomorphic-fetch": "^3.0.0",
    "json-beautify": "^1.1.1",
    "jwt-decode": "^3.1.2",
    "moment": "^2.29.1",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-helmet": "^6.1.0",
    "react-redux": "^7.2.3",
    "redux": "^4.0.5",
    "redux-devtools-extension": "^2.13.9",
    "redux-logger": "^3.0.6",
    "ts-invariant": "^0.7.3"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^5.12.0",
    "babel-jest": "^26.6.3",
    "babel-preset-gatsby": "^1.3.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^26.6.3",
    "jest-html-reporter": "^3.3.0",
    "prettier": "2.2.1",
    "react-test-renderer": "^17.0.2"
  },
  "keywords": [
    "gatsby"
  ],
  "license": "0BSD",
  "scripts": {
    "build": "gatsby build --open",
    "develop": "gatsby develop",
    "format": "prettier --write \"**/*.{js,jsx,ts,tsx,json,md}\"",
    "start": "npm run develop",
    "serve": "gatsby serve",
    "clean": "gatsby clean",
    "test": "jest --watch --verbose --collectCoverage=false",
    "coverage": "jest --coverage"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/gatsbyjs/gatsby-starter-default"
  },
  "bugs": {
    "url": "https://github.com/gatsbyjs/gatsby/issues"
  }
}
