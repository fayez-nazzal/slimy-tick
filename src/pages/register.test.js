import React from 'react';
import {
  cleanup,
  fireEvent,
  render,
  waitFor,
  screen,
} from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import renderer from 'react-test-renderer';
import RegisterPage from './register';
import WrapRootElement from '../wrap-root-element';
import registerMocks from '../../__mocks__/register-mocks';

describe('register input elements', () => {
  let registerPage;
  const component = (
    <WrapRootElement
      element={(
        <MockedProvider mocks={registerMocks} addTypename={false}>
          <RegisterPage />
        </MockedProvider>
      )}
    />
  );

  beforeEach(() => {
    registerPage = render(component);
  });

  afterEach(() => {
    cleanup();
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('email input set value correctly', () => {
    const emailInput = registerPage.getByLabelText('email');

    fireEvent.change(emailInput, {
      target: {
        value: 'test',
      },
    });

    expect(emailInput.value).toMatch(/^test$/i);
  });

  it('displays an error when no email is entered', async () => {
    const loginButton = registerPage.getByTestId('register-button');

    fireEvent.click(loginButton);

    const emailErrorMessage = await waitFor(() => registerPage.getByText(/email must not be empty/i));

    expect(emailErrorMessage).toBeTruthy;
  });

  it('displays an error when no password is entered', async () => {
    const registerButton = registerPage.getByTestId('register-button');

    fireEvent.click(registerButton);

    const passwordErrorMessage = await waitFor(() => registerPage.getByText(/password must not be empty/i));

    expect(passwordErrorMessage).toBeTruthy;
  });

  it('displays an error when no email is entered', async () => {
    const emailInput = registerPage.getByLabelText('email');

    fireEvent.change(emailInput, {
      target: {
        value: 'wrong@email..com',
      },
    });

    const registerButton = registerPage.getByTestId('register-button');

    fireEvent.click(registerButton);

    const emailError = await waitFor(() => registerPage.getByText(/email not valid/i));

    expect(emailError).toBeTruthy();
  });

  it("displays error when passwords don't match", async () => {
    const emailInput = registerPage.getByLabelText('email');
    const passwordInput = registerPage.getByLabelText('password');
    const confirmPasswordInput = registerPage.getByLabelText(
      'confirm password',
    );

    fireEvent.change(emailInput, {
      target: {
        value: 'correct@email.com',
      },
    });

    fireEvent.change(passwordInput, {
      target: {
        value: 'valid password',
      },
    });

    fireEvent.change(confirmPasswordInput, {
      target: {
        value: 'valid.password',
      },
    });

    const registerButton = registerPage.getByTestId('register-button');

    fireEvent.click(registerButton);

    const noMatchError = await waitFor(() => registerPage.getByText(/passwords must match/i));

    expect(noMatchError).toBeTruthy();
  });

  it('show error for short passwords', async () => {
    const emailInput = registerPage.getByLabelText('email');
    const passwordInput = registerPage.getByLabelText('password');
    const confirmPasswordInput = registerPage.getByLabelText(
      'confirm password',
    );

    fireEvent.change(emailInput, {
      target: {
        value: 'correct@email.com',
      },
    });

    fireEvent.change(passwordInput, {
      target: {
        value: '123',
      },
    });

    fireEvent.change(confirmPasswordInput, {
      target: {
        value: '123',
      },
    });

    const registerButton = registerPage.getByTestId('register-button');

    fireEvent.click(registerButton);

    const lengthError = await waitFor(() => registerPage.getByText(/Pasword must have 6 or more characters/i));

    expect(lengthError).toBeTruthy();
  });

  it('register successfuly when correct info provided', async () => {
    Object.defineProperty(window, '___navigate', { value: jest.fn() });
    const emailInput = registerPage.getByLabelText('email');
    const passwordInput = registerPage.getByLabelText('password');
    const confirmPasswordInput = registerPage.getByLabelText(
      'confirm password',
    );

    fireEvent.change(emailInput, {
      target: {
        value: 'correct@email.com',
      },
    });

    fireEvent.change(passwordInput, {
      target: {
        value: 'valid password',
      },
    });

    fireEvent.change(confirmPasswordInput, {
      target: {
        value: 'valid password',
      },
    });

    const registerButton = registerPage.getByTestId('register-button');

    fireEvent.click(registerButton);

    const validErrors = await waitFor(() => screen.queryAllByText(/not valid/i));
    const emptyErrors = await waitFor(() => screen.queryAllByAltText(/must not be empty/i));
    const matchErrors = await waitFor(() => screen.queryAllByAltText(/must match/i));

    expect(validErrors.length).toBe(0);
    expect(emptyErrors.length).toBe(0);
    expect(matchErrors.length).toBe(0);
  });
});
