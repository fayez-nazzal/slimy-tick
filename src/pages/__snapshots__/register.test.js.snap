// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['register input elements matches the snapshot 1'] = `
<div
  className="MuiContainer-root makeStyles-container-2 MuiContainer-disableGutters MuiContainer-maxWidthLg"
>
  <div
    className="makeStyles-buttonsContainer-4"
  >
    <a
      className="makeStyles-link-8"
      href="/register"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-5 MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Register
        </span>
      </button>
    </a>
    <a
      className="makeStyles-link-8"
      href="/login"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-5 MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Login
        </span>
      </button>
    </a>
  </div>
  <div
    className="makeStyles-logoContainer-6"
  >
    <test-file-stub
      className="makeStyles-logo-7"
    />
  </div>
  <div
    className="MuiContainer-root makeStyles-innerContainer-3 MuiContainer-maxWidthSm"
  >
    <h1
      className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
    >
      Register to Slimy Tick
    </h1>
    <h2
      className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
    >
      To have your first productive day
    </h2>
    <form
      onSubmit={[Function]}
    >
      <h6
        className="MuiTypography-root makeStyles-error-11 makeStyles-error-20 MuiTypography-subtitle1 MuiTypography-colorError MuiTypography-displayBlock"
      />
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="email-input"
          id="email-input-label"
        >
          Email
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            aria-label="email"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            id="email-input"
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-13 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-15"
            >
              <span>
                Email
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <h6
        className="MuiTypography-root makeStyles-error-11 makeStyles-error-21 MuiTypography-subtitle1 MuiTypography-colorError MuiTypography-displayBlock"
      />
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="password-input"
          id="password-input-label"
        >
          Password
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            aria-label="password"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            id="password-input"
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-13 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-15"
            >
              <span>
                Password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <h6
        className="MuiTypography-root makeStyles-error-11 makeStyles-error-22 MuiTypography-subtitle1 MuiTypography-colorError MuiTypography-displayBlock"
      />
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="confirm-password-input"
          id="confirm-password-input-label"
        >
          Confirm password
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            aria-label="confirm password"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            id="confirm-password-input"
            name="confirmPassword"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-13 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-15"
            >
              <span>
                Confirm password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <h6
        className="MuiTypography-root makeStyles-error-11 makeStyles-error-23 MuiTypography-subtitle1 MuiTypography-colorError MuiTypography-displayBlock"
      />
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-1 MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
        data-testid="register-button"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Register
        </span>
      </button>
    </form>
  </div>
</div>
`;
